editor/src/document/document_message_handler.rs

```diff
@@ -226,6 +230,22 @@ impl DocumentMessageHandler {
 		})
 	}
 
+	pub fn bounding_boxes<'a>(&self, ignore_document: Option<&Vec<Vec<LayerId>>>, ignore_artboard: Option<LayerId>) -> Vec<[DVec2; 2]> {
+		[
+			self.visible_layers()
+				.filter(|path| ignore_document.map_or(true, |ignore_document| !ignore_document.iter().any(|ig| ig.as_slice() == *path)))
+				.filter_map(|path| self.graphene_document.viewport_bounding_box(path).ok()?)
+				.collect(),
+			self.artboard_message_handler
+				.artboard_ids
+				.iter()
+				.filter(|id| Some(**id) != ignore_artboard)
+				.filter_map(|path| self.artboard_message_handler.artboards_graphene_document.viewport_bounding_box(&[*path]).ok()?)
+				.collect::<Vec<[DVec2; 2]>>(),
+		]
+		.concat()
```

Comment:
would it be possible to return a chained iterator instead?

Reply:
Sure - forgot about that :)